### ISSUE TITLE
Create a method for returning a JSON response 

### ISSUE THREAD

**Issue by bschmalhofer** [RotherOSS/otobo#2327]
Currently several frontend modules have a helper method that creates a JSON response. This is an indication that this is a commonly used functionality that warrants its own method in the layout object.

```
$ grep -r 'sub _ReturnJSON' Kernel/
Kernel/Modules/AdminSystemConfigurationUser.pm:sub _ReturnJSON {
Kernel/Modules/AgentReferenceSearch.pm:sub _ReturnJSON {
Kernel/Modules/AdminSystemConfigurationGroup.pm:sub _ReturnJSON {
Kernel/Modules/AgentContactWDSearch.pm:sub _ReturnJSON {
```

**Comment by chaitanyak175** [comment]
Hey, I want to get involved in the opensource community how can I solve this issue. Please help as this will be my first open source contribution.

**Comment by bschmalhofer** [comment]
Hi @chaitanyak175,

I'm not sure whether OTOBO is the best suited project for an overall open source contribution. As a first step I suggest that you do an installation of OTOBO so that you can see whether your contribution is working. Take a look at https://doc.otobo.org/manual/installation/10.1/en/content/installation-docker.html and see  whether you can get started. Note that documentation fixes are welcome as well.

Best regards,
    Bernhard

**Comment by bschmalhofer** [comment]
Hi @chaitanyak175,

I already have added the method `JSONReply()`. But there are still some things to do:

- [x] Refactor the remaining `_ReturnJSON()` helper methods to use `JSONReply()` instead
- [x] investigate whether other frontend modules can make use of `JSONReply()` 

No need to limit the passed in data to Hashref, Arrayref, and String. Cpanel::JSON::XS already croaks.

**Comment by bschmalhofer** [comment]
Let's do this after #399 is merged into rel-11_0. This reduced the number of variants that must be tested.

**Comment by bschmalhofer** [comment]
A lot more frontend modules could be converted. But there is no good reason to do that now. Closing this issue.
